src/NServiceBus.Core/ObjectBuilder/LightInject/LightInject.g.cs

```diff
@@ -2512,12 +2580,12 @@ public ServiceContainer(Action<ContainerOptions> configureOptions)
             ConstructorSelector = new MostResolvableConstructorSelector(CanGetInstance, options.EnableOptionalArguments);
             constructionInfoProvider = new Lazy<IConstructionInfoProvider>(CreateConstructionInfoProvider);
             methodSkeletonFactory = (returnType, parameterTypes) => new DynamicMethodSkeleton(returnType, parameterTypes);
-#if NET452 || NETSTANDARD1_3 || NETSTANDARD1_6 || NETSTANDARD2_0 || NET472 || NETCOREAPP
+#if NET452 || NETSTANDARD1_3 || NETSTANDARD1_6 || NETSTANDARD2_0 || NET46 || NET472 || NETCOREAPP2_0 || NETCOREAPP2_1
```

Comment:
In #5913, I made a different adjustment to the ifdefs, so I'm wondering if that would be better?

Instead of just adding more options, I replaced `NET46` -> `NET472` and for `NETCOREAPP2_0` -> `NETCOREAPP`

Reply:
Either works, but I think I slightly prefer addition over amendment.

It also makes the note simpler 🙂 

> ...and NET472 and NETCOREAPP2_1 added to the conditional compilation conditions.

vs

> ...and with NET46 and NETCOREAPP2_0 replaced with NET472 and NETCOREAPP in the conditional compilation conditions.

Reply:
Yeah, addition also works, just wanted to bring to your attention that I had recently also had to make a change here, and had gone a different way in case that made more sense.

I do still wonder if we would be better served with going with more generic additions, `NETFRAMEWORK`, `NETCOREAPP` instead.

That way the correct code paths here would still be chosen if we updated target frameworks and these ifdefs weren't touched (which has happened already once).

Reply:
How about we just remove the conditionals altogether?

Reply:
The more changes we make in the file, the more work it is to understand those changes and apply them to future updates. Seems like doing something like removing the conditionals might be going too far?

Reply:
@bording I've amended the commit to use your suggestion of `NETFRAMEWORK` and `NETCOREAPP`, and I've added inline notes in the two places which are amended.

Reply:
@adamralph It seems that you only altered some of them. It that intentional?

Reply:
@bording yes, I only altered the minimum required to get it all working.

Reply:
I don't understand, it seems like they should all be altered.

Reply:
I literally did the absolute minimum to get the code to compile, which seems to align with your [previous comment](https://github.com/Particular/NServiceBus/pull/5950#discussion_r585707035):

> The more changes we make in the file, the more work it is to understand those changes and apply them to future updates.